docs/library/socket.rst

```diff
@@ -66,19 +66,6 @@ Tuple address format for ``socket`` module:
 Functions
 ---------
 
-.. function:: socket(af=AF_INET, type=SOCK_STREAM, proto=IPPROTO_TCP, /)
```

Comment:
The reason this is documented as a function is because that's how it's done in CPython.  And that's because there are multiple functions (in CPython) that create sockets, with various functionality.  So I think it's best to follow CPython here and leave it documented as a function (even though technically it's implemented as a class/type constructor, but that could change in the future).

Reply:
I did check against the Python source from 3.4 .. 3.9.  All these implement socket as `class socket(_socket.socket):` 
Also in the [3.9 documentation socket.socket](https://docs.python.org/3/library/socket.html#socket.socket) is now indeed documented as a class, thus following its implementation. 
after searchin I now find that this was a docbug in python as well: https://bugs.python.org/issue45772

Reply:
The CPython docfix did not change the location in the documentation as per this change. Happy to move it back to the same location if that makes more sense from a textual perspective.

Reply:
You are right, it is now a class in CPython (I was looking at an older version of CPy 3.10).

I think moving it down in the document is fine.